tests/yast2_gui/yast2_control_center.pm

```diff
@@ -82,20 +82,29 @@ sub start_online_update {
         select_console 'x11', await_console => 0;
     }
     assert_and_click 'yast2_control-center_online-update';
-    assert_screen [qw(yast2_control-center_update-repo-dialogue yast2_control-center_online-update_close)];
-    if (match_has_tag('yast2_control-center_update-repo-dialogue')) {
-        send_key 'alt-n';
-        assert_screen 'yast2_control-center_online-update_close';
-    }
+    my @tags = qw(yast2_control-center_update-repo-dialogue yast2_control-center_online-update_close yast2_control-center-ask_packagekit_to_quit);
+    do {
+        assert_screen \@tags;
+        wait_screen_change { send_key 'alt-n' } if match_has_tag('yast2_control-center_update-repo-dialogue');
+        # Let it kill PackageKit, in case it is running.
+        wait_screen_change { send_key 'alt-y' } if match_has_tag('yast2_control-center-ask_packagekit_to_quit');
```

Comment:
In case the pop-up takes a bit longer to disappear that would meant the same needle would apply again and it would press the same key combination again on the next loop iteration.

Reply:
I'm with @cwh42 here, we saw similar issue already. I would rather think how many iterations should we conduct, as if pop-up is there, we will timeout after 2 hours only (the risk of such scenario is rather low).

Reply:
I see, makes sense